File: 5kyu/Valid Parentheses/Typescript/src/solution.ts
export const validParentheses = (parens: string): boolean => {
    let brackets = 0;
    for (let i = 0; i < parens.length; ++i) {
        const c = parens.charAt(i);
        if (c === "(") {
            ++brackets;
        } else if (c === ")") {
            --brackets;
        }
        if (brackets < 0) {
            return false;
        }
    }

    return brackets === 0;
};

File: 5kyu/Valid Parentheses/C++/src/solution.cpp
// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
#include "solution.hpp"

using std::string;
using std::size_t;

bool validParentheses(const string& parens) noexcept
{
	auto brackets = size_t{ 0 };
	for (const auto c:  parens) {
		if (c == '(') {
			++brackets;
		}
		else if (c == ')') {
			if (brackets == 0) {
				return false;
			}

			--brackets;
		}
	}

	return brackets == 0;
}